saleor/discount/utils/order.py

```diff
@@ -251,3 +262,26 @@ def create_or_update_line_discount_objects_for_manual_discounts(lines_info):
 
     if discount_to_update:
         OrderLineDiscount.objects.bulk_update(discount_to_update, ["amount_value"])
+
+
+def update_unit_discount_reason_with_order_level_discounts(
+    lines: Iterable[OrderLine], order: Order
+):
+    # Manual discounts can be added without a reason provided. To not lose info about
+    # applied manual discount we add default reason
```

Comment:
Sorry, I don't understand your concerns. If you provide "TBD" as a reason it will be visible in `unitDiscountReason` field as "TBD", so we don't lose the information.